pymeasure/instruments/lakeshore/lakeshore421.py

```diff
@@ -0,0 +1,448 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2019 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, \
+    truncated_discrete_set
+
+import numpy as np
+from time import time, sleep
+
+
+class LakeShore421(Instrument):
+    """
+    Represents the Lake Shore 421 Gaussmeter and provides a high-level interface for interacting
+    with the instrument.
+    .. code-block:: python
+
+        gaussmeter = LakeShore421("COM1")
+        gaussmeter.unit = "T"               # Set units to Tesla
+        gaussmeter.auto_range = True        # Turn on auto-range
+        gaussmeter.fast_mode = True         # Turn on fast-mode
+
+
+    A delay of 50 ms is ensured between subsequent writes, as the instrument cannot correctly
+    handle writes any faster.
+    """
+
+    MULTIPLIERS = {1e3: 'k', 1: '', 1e-3: 'm', 1e-6: 'n'}
+    PROBE_TYPES = {"High Sensitivity": 0,
+                   "High Stability": 1,
+                   "Ultra-High Sensitivity": 2}
+    RANGES = [30e3, 3e3, 300, 30]  # in Gauss
+    RANGE_MULTIPLIER_PROBE = [1, 10, 0.01]
+    RANGE_MULTIPLIER_UNIT = {'G': 1, 'T': 1e-4}
+    UNITS = ['G', 'T']
+    WRITE_DELAY = 0.05
+
+    def __init__(self, resource_name, **kwargs):
+        super(LakeShore421, self).__init__(
+            resource_name,
+            "Lake Shore 421 Gaussmeter",
+            asrl=dict(
+                baud_rate=9600,
+                data_bits=7,
+                stop_bits=10,
+                parity=1,
+                read_termination='\r',
+                write_termination='\n',
+            ),
+            **kwargs
+        )
```

Comment:
According to updated documentation, I understand this way of setting defaults is suggested for instruments with multiple interfaces. For single interface, you could simply set the default with initialized keyword parameters.

Reply:
Ahh, yes, I've changed it